plugins/auth-backend/src/identity/MountedTokenIssuer.ts

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AnyJWK, TokenIssuer, TokenParams } from './types';
+import { promises as fs } from 'fs';
+import {
+  JWK,
+  importSPKI,
+  importPKCS8,
+  exportJWK,
+  SignJWT,
+  importJWK,
+} from 'jose';
+import { parseEntityRef } from '@backstage/catalog-model';
+import { AuthenticationError } from '@backstage/errors';
+import { KeyLike } from 'jose/dist/types/types';
+import { LoggerService } from '@backstage/backend-plugin-api';
+
+const MS_IN_S = 1000;
+const DEFAULT_ALGORITHM = 'ES256';
+const DEFAULT_KEY_ID = '1';
+
+export type Config = {
+  publicKeyFile: string;
+  privateKeyFile: string;
+  keyId?: string;
+  algorithm?: string;
+};
+
+export type Options = {
+  logger: LoggerService;
+  /** Value of the issuer claim in issued tokens */
+  issuer: string;
+  /** Path to the public key file in the SPKI format */
+  publicKeyFile: string;
+  /** Path to the matching private key in the PKCS#8 format */
+  privateKeyFile: string;
+  /** Expiration time of signing keys in seconds */
+  keyDurationSeconds: number;
```

Comment:
Just taking a quick scan here, but is this maybe the expiry duration of _tokens_ signed by the given key, rather than an expiry duration for the key itself?

Actually, I see you're reusing the name/description from the other token issuer implementation, maybe my concern is more with that one.

Reply:
Hey that's correct. I've used the same name, I think in the TokenFactory it means **both** how long the signed token is valid as well as how long the private key is valid. While in this case it only means how long the signed token is valid.

Maybe I should change it to tokenTimeToLive or something akin to that?

Reply:
That seems like a way better name to me 😄

Reply:
Definitely an improvement 👍

Reply:
Glad you like it!